src/main/java/org/testng/internal/Parameters.java

```diff
@@ -102,10 +102,14 @@
 
     mapping.put(BeforeMethod.class.getSimpleName(), beforeAfterMethod);
     mapping.put(AfterMethod.class.getSimpleName(), beforeAfterMethod);
-    mapping.put(Test.class.getSimpleName(), Arrays.asList(new Class<?>[] {ITestContext.class}));
+    mapping.put(Test.class.getSimpleName(), newList(ITestContext.class));
 
   }
 
+  private static List<Class<?>> newList(Class<?>...elements) {
```

Comment:
And what about 

`List<Class<?>> ctxTest = Arrays.<Class<?>>asList(ITestContext.class, XmlTest.class);`